### PR TITLE
Refine rotation strategy handling and tests

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- replace the optional rotation generic with a boxed `RotationStrategy` in the file worker and builder wiring
- update rotating handler construction to pass boxed strategies and keep defaults aligned
- add targeted unit tests covering builder options, worker rotation integration, and `FileRotationStrategy` edge cases

## Testing
- make fmt
- RUSTC_WRAPPER= make lint
- RUSTC_WRAPPER= make test

------
https://chatgpt.com/codex/tasks/task_e_68dde0dc8c608322b5e47788fe46c4ca

## Summary by Sourcery

Simplify rotation strategy API by replacing Option-based generics with boxed trait objects across builders and workers, and enhance test coverage for rotation integration and edge cases

Enhancements:
- Replace the generic optional rotation type with a boxed dyn RotationStrategy in BuilderOptions and handler wiring
- Update all builder and spawn_worker calls to pass boxed rotation strategies and maintain default no-op behavior

Tests:
- Add unit tests for BuilderOptions default and custom rotation handling with a start barrier
- Add integration test verifying build_from_worker invokes a custom CountingRotation strategy
- Add edge-case tests for FileRotationStrategy covering backup limits, disabled backups, and zero-byte disablement

### CODE REVIEW

File: rust_extension/src/handlers/file/tests.rs [hunk 1]
```diff
@@ -142,7 +232,7 @@ fn worker_writes_record_when_rotation_fails() {
         flush_interval: 1,
         overflow_policy: OverflowPolicy::Block,
     };
-    let options = BuilderOptions::<SharedBuf, FailingRotation>::new(Some(FailingRotation), None);
+    let options = BuilderOptions::<SharedBuf>::new(Box::new(FailingRotation), None);
```
[the remark this answers is not in the document]
Comment (leynos) [reply rust_extension/src/handlers/file/tests.rs]: Has this now been resolved?
